src/module/actor/SR5Actor.ts

```diff
@@ -2294,4 +2296,41 @@ export class SR5Actor<SubType extends Actor.ConfiguredSubType = Actor.Configured
         await StorageFlow.deleteStorageReferences(this);
         return super._preDelete(...args);
     }
+
+    async deleteItem(item: SR5Item | SR5ActiveEffect) {
+        if (item instanceof SR5Item) {
+            if (ActorOwnershipFlow._isOwnerOfItem(this, item)) {
+                if (item._id) {
+                    await this.deleteEmbeddedDocuments('Item', [item._id]);
+                } else {
+                    console.error("Could not find _id field item, cannot delete an unowned item by the actor");
+                }
+            }
+        }
+    }
+
+    getSource() {
+        return this.system.description.source ?? '';
+    }
+
+    get hasSource(): boolean {
+        return !!this.getSource()
+    }
+
+    async openSource() {
+        const source = this.getSource();
+        await LinksHelpers.openSource(source);
+    }
+
+    favorites() {
+        return new Set(this.system.favorites);
+    }
+
+    hiddenItems() {
+        return new Set(this.system.hidden_items);
+    }
+
+    hasItemOfType<ST extends readonly Item.ConfiguredSubType[]>(...types: ST): this is SR5Item<ST[number]> {
```

Comment:
`this is` works to automatically narrow the type you are working on. By this, you are saying that this actor is also an item, so it will always narrow the actor to `never`.

what you should do here is:
`hasItemOfType(...types: Item.ConfiguredSubType[])`